simd/unit_tests/include/SIMDTesting_Ops.hpp

```diff
@@ -119,4 +119,183 @@ class shift_left {
   }
 };
 
+class cbrt_op {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+#if defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_INTEL_LLVM)
+    using abi_type = typename T::abi_type;
+    if constexpr (!std::is_same_v<abi_type,
+                                  Kokkos::Experimental::simd_abi::scalar>)
+      return Kokkos::cbrt(a);
+#endif
+    T result(a);
+    for (std::size_t i = 0; i < result.size(); ++i) {
+      result[i] = Kokkos::cbrt(result[i]);
+    }
+    return result;
+  }
+  template <typename T>
+  auto on_host_serial(T const& a) const {
+    return Kokkos::cbrt(a);
+  }
+};
+
+class exp_op {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+#if defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_INTEL_LLVM)
+    using abi_type = typename T::abi_type;
+    if constexpr (!std::is_same_v<abi_type,
+                                  Kokkos::Experimental::simd_abi::scalar>)
+      return Kokkos::exp(a);
+#endif
+    T result(a);
+    for (std::size_t i = 0; i < result.size(); ++i) {
+      result[i] = Kokkos::exp(result[i]);
+    }
+    return result;
+  }
+  template <typename T>
+  auto on_host_serial(T const& a) const {
+    return Kokkos::exp(a);
+  }
+};
+
+class log_op {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+#if defined(KOKKOS_COMPILER_INTEL) || defined(KOKKOS_COMPILER_INTEL_LLVM)
+    using abi_type = typename T::abi_type;
+    if constexpr (!std::is_same_v<abi_type,
+                                  Kokkos::Experimental::simd_abi::scalar>)
+      return Kokkos::log(a);
+#endif
+    T result(a);
+    for (std::size_t i = 0; i < result.size(); ++i) {
+      result[i] = Kokkos::log(result[i]);
+    }
+    return result;
+  }
+  template <typename T>
+  auto on_host_serial(T const& a) const {
+    return Kokkos::log(a);
+  }
+};
+
+class hmin {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+    return Kokkos::Experimental::hmin(a);
```

Comment:
Why are these qualified with `Kokkos::Experimental::` and other only with `Kokkos::`?

Reply:
The ones used in `on_host()` and `on_device()` are meant to exercise functions newly implemented in simd backend. So they are qualified with `Kokkos::Experimental::` to invoke simd functions.
The others, used in `on_host_serial()` and `on_device_serial()` are not necessarily meant to exercise simd functions. They are meant to be serial implementations using existing Kokkos functions to test the output value against the new simd functions.

Reply:
But then shouldn't L174 for instance be qualified with `Kokkos::Experimental::`?

Reply:
You are right. Fixed namespaces for `cbrt`, `exp` and `log` to have them invoke correct simd functions. [fc10c05](https://github.com/kokkos/kokkos/pull/6276/commits/fc10c05fa22a4ef06f632ea802c3af750dddddaa)